sys/tools/syscalls/bsdio.lua

```diff
@@ -0,0 +1,144 @@
+--
+-- SPDX-License-Identifier: BSD-2-Clause
+--
+-- Copyright (c) 2024 Tyler Baxter <agge@FreeBSD.org>
+--
+--
+-- Thanks to Kyle Evans for his makesyscall.lua in FreeBSD which served as
+-- inspiration for this, and as a source of code at times.
+--
+-- SPDX-License-Identifier: BSD-2-Clause-FreeBSD
+--
+-- Copyright (c) 2019 Kyle Evans <kevans@FreeBSD.org>
+--
+
+util = require("util")
+
+local bsdio = {}
```

Comment:
I'm planning on renaming to `generator`, since much of its use is binding to a file and writing. I'm also considering splitting off the macros into macros.lua (or re-evaluating if they provide value). Let me know if this sounds better.